src/server/v2.0/handler/ldap.go

```diff
@@ -0,0 +1,109 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/goharbor/harbor/src/common/rbac"
+	"github.com/goharbor/harbor/src/common/security"
+	ctlLdap "github.com/goharbor/harbor/src/controller/ldap"
+	"github.com/goharbor/harbor/src/lib/errors"
+	ldapModel "github.com/goharbor/harbor/src/pkg/ldap/model"
+	"github.com/goharbor/harbor/src/server/v2.0/models"
+	"github.com/goharbor/harbor/src/server/v2.0/restapi/operations/ldap"
+)
+
+type ldapAPI struct {
+	BaseAPI
+	ctr ctlLdap.Controller
+}
+
+func newLdapAPI() *ldapAPI {
+	return &ldapAPI{ctr: ctlLdap.Ctl}
+}
+
+func (l *ldapAPI) PingLdap(ctx context.Context, params ldap.PingLdapParams) middleware.Responder {
+	if err := l.RequireSystemAccess(ctx, rbac.ActionRead, rbac.ResourceConfiguration); err != nil {
+		return l.SendError(ctx, err)
+	}
+	basicCfg := ldapModel.LdapConf{
+		URL:            params.Ldapconf.LdapURL,
+		BaseDn:         params.Ldapconf.LdapBaseDn,
+		SearchDn:       params.Ldapconf.LdapSearchDn,
+		Filter:         params.Ldapconf.LdapFilter,
+		SearchPassword: params.Ldapconf.LdapSearchPassword,
+		UID:            params.Ldapconf.LdapUID,
+		Scope:          int(params.Ldapconf.LdapScope),
+		VerifyCert:     params.Ldapconf.LdapVerifyCert,
+	}
+	payload := &models.LdapPingResult{}
+	suc, err := l.ctr.Ping(ctx, basicCfg)
+	payload.Success = suc
+	if err != nil {
+		payload.Message = fmt.Sprintf("error: %v", err)
+	}
+	return ldap.NewPingLdapOK().WithPayload(payload)
+}
+
+func (l *ldapAPI) SearchLdapUser(ctx context.Context, params ldap.SearchLdapUserParams) middleware.Responder {
+	secCtx, ok := security.FromContext(ctx)
+	if !ok {
+		return l.SendError(ctx, errors.UnauthorizedError(errors.New("security context not found")))
+	}
+	if !secCtx.IsSysAdmin() {
+		return l.SendError(ctx, errors.ForbiddenError(errors.New("only admin has this authority")))
+	}
```

Comment:
Could we introduce a new rbac resource , eg LDAPUser, and use `l.RequireSystemAccess(ctx, rbac.ActionList, rbac.LDAPUser )` to check the permission of this API so that the system robot account has the ability to request this API?

In general, we don't check the SecurityContext must be the sysadmin in the API handler after v2.2.

Reply:
Add LDAPUser resource and read/create action